### DISCUSSION TITLE
Pages to App router incremental migration - how?

### DISCUSSION THREAD

**Discussion by craig-ryan-sn** [vercel/next.js#58163] [Help]
### Summary

I've been spending considerable time following the guide which suggests incremental upgrade to app router is possible. 

As with any normal application beyond 'hello world' we have many common components. I'm seeing two main blockers at present, with not idea how many will follow since half my app won't load.

- useRouter is imported from various components (next/router vs next/navigation) 
- next-i18next is imported from many components, which only work via pages/

There is no incremental migration I can see possible here. The routers are incompatible (am I correct - next/navigation is not backward compatible?) and the i18n migration blog only works if NO app imports pull in components which import next-i18next which is virtually impossible to correct without a lot of code duplication.

I'm looking for guidance here, to help confirm I'm facing the likelihood of a all-or-nothing complete migration to app router?

### Additional information

I'm referring to this blog https://locize.com/blog/next-app-dir-i18n/

**Comment by Henkedanne** [comment]
https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#step-5-migrating-routing-hooks

**Reply by FezVrasta** [reply]
I think the elephant in the room is the difference between `next/router` and `next/navigation`.

If I have a component that needs `useRouter` there's no way to use it on both a page route and an app one?
